falcon-solver/README.md

```diff
@@ -1,25 +1,36 @@
-# CausalSolver
+# Falcon Solver
 
-CausalSolver uses symbolic constraint solving to order a set of distributed events according to their *happens-before* relationship [1]. There exists a happens-before relationship between two events **A** and **B** if **A** causally precedes **B**. Any two events **A'** and **B'** that are not ordered by a happens-before dependency are said to be *concurrent*. 
+Falcon Solver uses symbolic constraint solving to order a set of distributed events according to their *happens-before* relationship [1]. There exists a happens-before relationship between two events **A** and **B** if **A** causally precedes **B**. Any two events **A'** and **B'** that are not ordered by a happens-before dependency are said to be *concurrent*. 
 
-### Requirements:
-* Z3 SMT solver
+In more detail, Falcon Solver:
+1. Receives an input file with a set of events in JSON format;
+2. Builds an SMT constraint model that contains symbolic variables representing the events' logical clocks and constraints encoding the happens-before relationships between them;
+3. Solves the constraint model to obtain the logical clocks for the events that satisfy the causal dependencies.
 
-# How to use
+**References:**
 
-CausalSolver i) receives an input file with a set of events in JSON format, ii) builds an SMT constraint model that encodes child causality in terms of symbolic constraint, and iii) solves the constraint model to obtain a logical timestamp for each child, which reflects its potential dependency on previous events.
+[1] L. Lamport, “Time clocks, and the ordering of events in a distributed system”, Commun. ACM, vol. 21, pp. 558–565, July 1978.
 
-(Assuming `$HOME=<path/to/causalSolver>`)
-* Compile: `$ mvn package`
-* Run: `$ java -jar $HOME/target/causalSolver-1.0-SNAPSHOT-jar-with-dependencies.jar`
+## Requirements
+* [Z3 Theorem Prover](https://github.com/Z3Prover/z3) (folder `lib` already contains Z3 v4.4.1 for MacOS)
```

Comment:
I would discard the `z3` bin file from this repository.

Leave a `.gitkeep` file there, instead.